### PR TITLE
DOC-167 updated based on review and added unit test cases

### PR DESCRIPTION
**Pull request by mani0201**

### CODE REVIEW

File: datasafe-rest-impl/src/test/java/de/adorsys/datasafe/rest/impl/controller/UserControllerTest.java [hunk 1]
```diff
@@ -43,19 +47,52 @@ void createUserTest() {
         verify(userProfile).registerUsingDefaults(any());
     }
 
+    @SneakyThrows
+    @Test
+    void createDuplicateUserTest() {
+
+        when(dataSafeService.userProfile().userExists(new UserID(TEST_USER))).thenReturn(true);
+
+        Assertions.assertThrows(NestedServletException.class, () -> {
```
Comment (valb3r) [inline datasafe-rest-impl/src/test/java/de/adorsys/datasafe/rest/impl/controller/UserControllerTest.java]:
```
put("/user")
                .contentType(MediaType.APPLICATION_JSON)
                .accept(MediaType.APPLICATION_JSON)
                .characterEncoding("UTF-8")
                .content("{ \"userName\" : \"" + TEST_USER + "\" , \"password\" : \"" + TEST_PASS + "\" }")
.header("token", token)
```
Can be extracted to separate function so code won't be duplicated - it is duplicated within happy-path-and exceptional path

File: datasafe-rest-impl/src/test/java/de/adorsys/datasafe/rest/impl/controller/UserControllerTest.java [hunk 2]
```diff
@@ -43,19 +47,52 @@ void createUserTest() {
...
+                    .content("{ \"userName\" : \"" + TEST_USER + "\" , \"password\" : \"" + TEST_PASS + "\" }")
+                    .header("token", token)
+            );
+        });
+    }
 
     @SneakyThrows
     @Test
     void deleteUserTest() {
+        if(userProfile.userExists(new UserID(TEST_USER))) {
```
Comment (valb3r) [inline datasafe-rest-impl/src/test/java/de/adorsys/datasafe/rest/impl/controller/UserControllerTest.java]:
```
delete("/user")
                    .contentType(MediaType.APPLICATION_JSON)
                    .accept(MediaType.APPLICATION_JSON)
                    .header("user", TEST_USER)
                    .header("password", TEST_PASS)
                    .header("token", token)
```
Can be extracted to separate function so code won't be duplicated - it is duplicated within happy-path-and exceptional path

File: datasafe-rest-impl/src/test/java/de/adorsys/datasafe/rest/impl/controller/UserControllerTest.java (hunk 1 above)
Comment (mani0201) [reply datasafe-rest-impl/src/test/java/de/adorsys/datasafe/rest/impl/controller/UserControllerTest.java]: Changes are updated.

**Comment by mani0201** [comment]
Due to Fork branch not able to run the trevis (not successful), as per Maksym comments, created own branch as DOC-167 and pushed the code and merged with develop branch.

**Pull request closed without merging**